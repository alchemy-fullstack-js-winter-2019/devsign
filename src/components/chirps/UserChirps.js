import React from 'react';
import PropTypes from 'prop-types';
import Chirp from './Chirp';
import { Main } from './UserChirpsStyles';

export default function UserChirps({ chirps }) {
  const listOfChirps = chirps.map((chirp, i) => {
    return <li key={i}><Chirp chirp={chirp.chirp} handle={chirp.handle} profileImg={chirp.profileImg} /></li>;
  });
  return (
    <Main>
      <ul>
        {listOfChirps}
      </ul>
    </Main>
  );
}

UserChirps.propTypes = {
  chirps: PropTypes.array
};
