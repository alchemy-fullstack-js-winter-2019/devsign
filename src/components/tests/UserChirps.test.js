import React from 'react';
import renderer from 'react-test-renderer';
import UserChirps from '../chirps/UserChirps';

describe('UserChirps', () => {
  it('matches a snapshot', () => {
    const chirps = [{
      chirp: 'this is a chirp'
    }];

    const tree = renderer.create(
      <UserChirps chirps={chirps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
