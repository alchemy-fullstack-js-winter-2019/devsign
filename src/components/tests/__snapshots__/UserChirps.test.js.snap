// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChirps matches a snapshot 1`] = `
<main
  className="sc-bwzfXH locBlQ"
>
  <ul>
    <li>
      <section
        className="sc-bdVaJa jaDQod"
      >
        <img
          alt="prof-pic"
          height="40px"
        />
        <div
          className="info"
        >
          <h4
            className="name"
          >
            UserName
          </h4>
          <p
            className="handle"
          />
          <p
            className="body"
          >
            this is a chirp
          </p>
        </div>
      </section>
    </li>
  </ul>
</main>
`;
